### DISCUSSION TITLE
How to upload non cypress test results to Cypress cloud

### DISCUSSION THREAD

**Discussion by leonid-deriv** [cypress-io/cypress#28017] [Questions and Help]
I need to upload test run results from backend tests to Cypress cloud.  I cannot find this information.  Can you point me to the right direction? Thank you

**Comment by portyu9** [comment]
Hi @leonid-deriv 

There is currently no supported Cypress Cloud API for uploading arbitrary **non-Cypress/backend test results** into Cypress Cloud as Cypress test runs.

Cypress Cloud test results are created when the Cypress runner executes tests with recording enabled, for example:

```bash
cypress run --record --key ...
```

Cypress does provide a **Data Extract API**, but that API works in the opposite direction: it lets Enterprise users retrieve/export test data that is already in Cypress Cloud. It is not an ingestion API.

So for backend tests you would need to keep their results in your CI/reporting system or another reporting platform.

If some of those backend checks can appropriately be expressed as Cypress API tests, running them as Cypress tests would allow those Cypress results to be recorded to Cloud, but Cypress Cloud cannot currently import an arbitrary external test-results file and turn it into a run.

Please mark this answer as accepted if it helped, thank you.
